docs/src/main/tut/docs/cassandra/README.md

```diff
@@ -4,6 +4,265 @@ title: Cassandra
+
+  def readOption[A](resultSet: ResultSet)(implicit FR: FromReader[A]): FS[Option[A]]
+
+  def readList[A](resultSet: ResultSet)(implicit FR: FromReader[A]): FS[List[A]]
+
+}
+```
+
+## Freatures
+
+### Low level Queries
+The Low-Level API allows us to define queries with placeholders for query constraints, which can be 
+later bound to the final values, running the query in a safe mode, preventing CQL injection. 
+Frees-Cassandra checks the generated query is a valid one at compile time, throwing errors in case 
+any input value doesn't match with the expected data type. 
+
+Let's see how we can use it:
+
+```
+for {
+  preparedStatement <- sessionAPI.prepare("INSERT INTO users (id, name) VALUES (?, ?)")
+  boundStatement    <- bindValues(preparedStatement)
+  _                 <- sessionAPI.executeStatement(boundStatement)
+  user <- sessionAPI.executeWithMap(
+    s"SELECT id, name FROM users WHERE id = ?",
+    Map("id" -> newUser.id)) map { rs =>
+    val row = rs.one()
+    User(row.getUUID(0), row.getString(1))
+  }
+} yield user
+
+```
+
+Note that we are using a bind function, which is defined as follows:
+
+```
+def bindValues(st: PreparedStatement)(
+    implicit c1: ByteBufferCodec[UUID],
+    c2: ByteBufferCodec[String]): FreeS[F, BoundStatement] =
+  List(
+    statementAPI.setValueByName(_: BoundStatement, "id", newUser.id, c1),
+    statementAPI.setValueByName(_: BoundStatement, "name", newUser.name, c2))
+    .foldLeft[FreeS[F, BoundStatement]](statementAPI.bind(st)) { (freeS, func) =>
+      freeS.flatMap(boundSt => func(boundSt))
+    }
+```
+
+
+### String Interpolator
+Frees-Cassandra Query Interpolator allows us to write a raw query and validate it against a 
+previously defined schema. 
+
+To do so, we need to first define an annotated trait specifying a path to the file containing the
+cassandra keyspaces & tables schemas. 
+
+Example
+
+```  @SchemaFileInterpolator("/schema.cql")
+     trait DummySchemaInterpolator
+```
+
+This annotation will be expanded via macros to implement a companion object defining a 
+schemaValidator as well as a cql string interpolator method. Now we can import this 
+DummySchemaInterpolator and take advantage of this macro generated utilities [1](#1).  
+
+```
+  import DummySchemaInterpolator._
+
+  val insertUserTask: FreeS[F, ResultSet] =
+    cql"INSERT INTO demodb.users (id, name) VALUES ($uuid, 'Username');".asResultSet[F]()
+  val getUserTask: FreeS[F, ResultSet] =
+    cql"SELECT id, name FROM demodb.users WHERE id = $uuid".asResultSet[F]()
+```
+
+At the end, what we are getting with `FreeS[F, ResultSet]` is a `Free[FreeApplicative[F, ?], A]` 
+which can be monadically composed as follows:
+
+```
+for {
+      _             <- insertUserTask
+      userResultSet <- getUserTask
+    } yield userResultSet
+```
+
+Full String Interpolator example is availiable at [Github Frees-Cassandra-Example](https://github.com/frees-io/freestyle-cassandra-example)
```

Comment:
It may be useful to include that under a section in the docs. The different libs like Kafka and Cassandra will eventually have many sections an can potentially grow beyond one menu entry at frees.io since we plan to considerably expand in their functionality and what they can do. Perhaps bringing a couple or one example from there here would be useful for users to understand how to use the interpolator.